src/workers/parser/utils/contains.js

```diff
@@ -3,12 +3,30 @@
 import type { AstLocation, AstPosition } from "../types";
 import type { Node } from "babel-traverse";
 
+function startsBeforeColumn(a: AstLocation, b: AstPosition) {
+  if (!(a.start.column && b.column)) {
+    return true;
+  }
+  return a.start.column <= b.column;
+}
+
+function endsAfterColumn(a: AstLocation, b: AstPosition) {
+  if (!(a.end.column && b.column)) {
+    return true;
+  }
+  return a.end.column >= b.column;
+}
+
 export function containsPosition(a: AstLocation, b: AstPosition) {
-  const startsBefore =
-    a.start.line < b.line ||
-    (a.start.line === b.line && a.start.column <= b.column);
-  const endsAfter =
-    a.end.line > b.line || (a.end.line === b.line && a.end.column >= b.column);
+  let startsBefore = a.start.line < b.line;
+  if (!startsBefore && a.start.line === b.line) {
+    startsBefore = startsBeforeColumn(a, b);
+  }
+
+  let endsAfter = a.end.line > b.line;
+  if (!endsAfter && a.end.line === b.line) {
+    endsAfter = endsAfterColumn(a, b);
+  }
```

Comment:
@jasonLaster I was thinking the following as well (but seems not really different from the current one):
```js
function startsBefore(a: AstLocation, b: AstPosition) {
  let startsBefore = a.start.line < b.line;
  if (a.start.line === b.line && a.start.column && b.column) {
    startsBefore = a.start.column <= b.column;
  }
  return startsBefore;
}

function endsAfter(a: AstLocation, b: AstPosition) {
  // The similar codes as startsBefore
}

export function containsPosition(a: AstLocation, b: AstPosition) {
  return startsBefore(a, b) && endsAfter(a, b);
}
```
Please let me know which is better you think, thanks

Reply:
I like it better :) 

```js
function startsBefore(a: AstLocation, b: AstPosition) {
  if (a.start.line === b.line && a.start.column && b.column) {
    return a.start.column <= b.column;
  }
  
  return a.start.line < b.line;
}

```

it's easier for me to read